cats/cats-redis/src/main/java/com/netflix/spinnaker/cats/redis/cache/RedisCacheOptions.java

```diff
@@ -0,0 +1,323 @@
+    public static Builder builder() {
+        return new Builder();
+    }
+    private final int maxMsetSize;
+    private final int maxMgetSize;
+    private final int maxHmgetSize;
+    private final int maxHmsetSize;
+    private final int maxSaddSize;
+    private final int maxDelSize;
+    private final int maxPipelineSize;
+    private final int scanSize;
+    private final int maxMergeBatchSize;
+    private final int maxEvictBatchSize;
+    private final int maxGetBatchSize;
+    private final boolean hashingEnabled;
+
+    private static int posInt(String name, int value) {
+        Preconditions.checkArgument(value > 0, "%s must be a positive integer (%s)", name, value);
+        return value;
+    }
+
+    private static int posEven(String name, int value) {
+        Preconditions.checkArgument(value > 0 && value % 2 == 0, "%s must be a positive even integer (%s)", name, value);
+        return value;
+    }
+
+    public RedisCacheOptions(int maxMsetSize, int maxMgetSize, int maxHmgetSize, int maxHmsetSize, int maxSaddSize, int maxDelSize, int maxPipelineSize, int scanSize, int maxMergeBatchSize, int maxEvictBatchSize, int maxGetBatchSize, boolean hashingEnabled) {
+        this.maxMsetSize = posEven("maxMsetSize", maxMsetSize);
+        this.maxMgetSize = posInt("maxMgetSize", maxMgetSize);
+        this.maxHmgetSize = posInt("maxHmgetSize", maxHmgetSize);
+        this.maxHmsetSize = posInt("maxHmsetSize", maxHmsetSize);
+        this.maxSaddSize = posInt("maxSaddSize", maxSaddSize);
+        this.maxDelSize = posInt("maxDelSize", maxDelSize);
+        this.maxPipelineSize = posInt("maxPipelineSize", maxPipelineSize);
+        this.scanSize = posInt("scanSize", scanSize);
+        this.maxMergeBatchSize = posInt("maxMergeBatchSize", maxMergeBatchSize);
+        this.maxEvictBatchSize = posInt("maxEvictBatchSize", maxEvictBatchSize);
+        this.maxGetBatchSize = posInt("maxGetBatchSize", maxGetBatchSize);
+        this.hashingEnabled = hashingEnabled;
+    }
+
+    public int getMaxMsetSize() {
+        return maxMsetSize;
+    }
+
+    public int getMaxMgetSize() {
+        return maxMgetSize;
+    }
+
+    public int getMaxHmgetSize() {
+        return maxHmgetSize;
+    }
+
+    public int getMaxHmsetSize() {
+        return maxHmsetSize;
+    }
+
+    public int getMaxSaddSize() {
+        return maxSaddSize;
+    }
+
+    public int getMaxDelSize() {
+        return maxDelSize;
+    }
+
+    public int getMaxPipelineSize() {
+        return maxPipelineSize;
+    }
+
+    public int getScanSize() {
+        return scanSize;
+    }
+
+    public int getMaxMergeBatchSize() {
+        return maxMergeBatchSize;
+    }
+
+    public int getMaxEvictBatchSize() {
+        return maxEvictBatchSize;
+    }
+
+    public int getMaxGetBatchSize() {
+        return maxGetBatchSize;
+    }
+
+    public boolean isHashingEnabled() {
+        return hashingEnabled;
+    }
+
+    public static class Builder {
```

Comment:
this is set up to enable builder but also exposes getter/setter so that `@ConfigurationProperties` binding works